validate.swift

```diff
@@ -125,6 +125,32 @@ extension Result where Success == Void {
 
 // MARK: Functions
 
+/**
+ Fetch the default branch for a given GitHub repository.
+ - Parameter userName: Repository owner
+ - Parameter repositoryName: Repository name
+ - Returns: default branch name if successful; otherwise `noResult` or a `decodingError`.
+ */
+func getGitHubDefaultBranch(for userName: String, repositoryName: String) -> Result<String, PackageError> {
+  struct Repository: Decodable {
+    var default_branch: String  // don't bother with re-casing this for now...
+  }
+  let sema = DispatchSemaphore(value: 0)
+  let apiURL = URL(string: "https://api.github.com/repos/\(userName)/\(repositoryName)")!
+  print("apiURL: \(apiURL.absoluteString)")
```

Comment:
Same comment here as with the other PR. Is this a debug print statement?

Reply:
Whoops - fixed

Reply:
Let me avoid that force upwrap as well

Reply:
I've marked this as Draft because I saw that the calling method wasn't actually using the error in the result. I can either

a) make `getGitHubDefaultBranch` return `String?` and then return with a new error `.failure(.invalidDefaultBranch)` from the calling function `getPackageSwiftURL`

or

b) properly unpack and handle the error details in `getPackageSwiftURL`, keeping the `Result` on `getPackageSwiftURL`

The first one will be a smaller overall diff, although it'll discard some error details. Which would you prefer?

Reply:
Not sure I was looking through what I do here:
https://github.com/SwiftPackageIndex/PackageListValidator/blob/release/0.0.1/Sources/PackageListValidator/Controllers/GitHubDefaultBranchQuery.swift

Just to let you know, the biggest reason this will throw an error is when you run the validation on the entire list and hit the API limit. That's what I've seen.

IMO, in that case, I'd want the app to either assume "master" or quit.

Reply:
Yeah, that'd be an issue. However my understanding was that you'd use this script to validate a handful of packages you're submitting? Right now that just errors out irrevocably with dependencies that don't have a default `master` branch and there's nothing you can do about it (I think?).

An alternative is to re-try in case the outdated tools error pops up – although that's going to be quite a bit more complicated.

What do you think, is it worth pursuing this?

Reply:
Most of the time yes.

If no default branch then assume "master" otherwise we should be notified of any other errors as issues with the script as opposed to issues with the Package for now. I would not do any sort of re-try at this point.

Does that make sense?

Reply:
Sure! Updated :)